Refactor: [M3-6277] Notification center MUI Migration

abailly-akamai opened the pull request:
## Description 📝
Mui migration for the notification center (feature)

**Note**: You'll notice some tightening of white space in this refactor aimed to condense content a little

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-05-23 at 2 27 24 PM](https://github.com/linode/manager/assets/130582365/8f6797bc-5258-4f4f-b575-cdc19bdb5f0c) | ![Screenshot 2023-05-24 at 10 38 09 AM](https://github.com/linode/manager/assets/130582365/7756cf51-f38a-4a8d-b7eb-619ec16cfa52) |

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-05-23 at 2 28 15 PM](https://github.com/linode/manager/assets/130582365/a7510646-eed2-497b-8992-d77f3094c10f) | ![Screenshot 2023-05-24 at 10 38 32 AM](https://github.com/linode/manager/assets/130582365/dd934551-6946-419b-a730-c361dc303bb6) |

| MSW  |   |
| ------- | ------- |
| ![Screenshot 2023-05-24 at 10 38 44 AM](https://github.com/linode/manager/assets/130582365/c5a8e297-478c-43e4-8172-dcd7f1b44935) |


## How to test 🧪
1. Test the various scenarii related to the notification center (standard, with a progress bar and entering a migration for instance, and any other case you're familiar with)

jaalah-akamai reviewed (approved):
Thanks for adding `Button/LinkStyledButton.ts` - looks good

jdamore-linode reviewed (approved):
Nice! Everything seems to be working fine, and I like the borders/whitespace tweaks

hana-linode reviewed (commented):
Personally, I think the design looks cleaner without the addition of light borders. (See #4 of https://medium.com/refactoring-ui/7-practical-tips-for-cheating-at-design-40c736799886 for more info)

mjac0bs commented on packages/manager/src/features/NotificationCenter/NotificationSection.tsx:
```diff
@@ -208,43 +166,80 @@ const ContentBody: React.FC<BodyProps> = React.memo((props) => {
+}));
+
+const StyledHeader = styled('div', {
+  label: 'StyledHeader',
+})(({ theme }) => ({
+  alignItems: 'center',
+  borderBottom: `solid 1px ${theme.borderColors.borderTypography}`,
+  display: 'flex',
+  justifyContent: 'space-between',
+  paddingBottom: theme.spacing(),
+}));
+
+const StyledLoadingContainer = styled('div', {
+  label: 'StyledLoadingContainer',
+})(() => ({
+  display: 'flex',
+  justifyContent: 'center',
+}));
+
+const StyledNotificationItem = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  fontSize: '0.875rem',
+  justifyContent: 'space-between',
+  marginTop: theme.spacing(),
+  width: '100%',
+  '& p': {
+    color: theme.textColors.headlineStatic,
+    lineHeight: '1.25rem',
+  },
+}));
+
+const StyledCaret = styled(KeyboardArrowDown)(({ theme }) => ({
+  color: theme.palette.primary.main,
+  marginLeft: theme.spacing(),
+}));
+
+const StyledEmptyMessage = styled(Typography)(({ theme }) => ({
+  marginTop: theme.spacing(),
+  marginBottom: theme.spacing(2.5),
+}));
```
Can we move these to their own file, [since this component is almost 200 lines long](https://github.com/linode/manager/blob/develop/docs/development-guide/02-component-structure.md#styles)?

mjac0bs commented on packages/manager/src/components/GravatarByUsername.tsx:
```diff
@@ -1,18 +1,18 @@
-    borderRadius: '50%',
-  },
-});
+const StyledAvatar = styled(Avatar, {
+  label: 'StyledGravatarByUsername',
+})(() => ({
+  height: 28,
+  width: 28,
+  borderRadius: '50%',
+}));
```
Can we move this to the bottom of this file, following [our coding standards for styles](https://github.com/linode/manager/blob/develop/docs/development-guide/02-component-structure.md#styles)?

mjac0bs reviewed (commented):
I don't want to start an all out border war, but I am curious to hear what UX decides on the light borders. I lean towards agreeing with @hana-linode, but if we're going to include them, we should fix this hover state color that seems to be impacted by margin or padding:

![Screenshot 2023-05-23 at 2 11 57 PM](https://github.com/linode/manager/assets/114685994/8fbfd863-d895-4f4a-a71e-9a1286ff50ef)

I think it is worth noting that by adding the borders to events list, we're straying more from our styling of the notifications list (turned on the MSW to see this). Not to say that our existing styling of notifications shouldn't be improved (I've always found the notifications to be too close together and we don't apply any hover states, either).

![Screenshot 2023-05-23 at 2 03 07 PM](https://github.com/linode/manager/assets/114685994/1847168f-1470-4472-ac55-359272f1692d)

abailly-akamai commented:
@hana-linode & @mjac0bs thanks for sharing! It's a healthy conversation, no fear of a war here. Valid points all around.

I added a fix while you were commenting this which lightens them to be more standard with how we use them (also removed the padding issue) and will tighten things a bit more in a subsequent commit, especially with the notification section as @mjac0bs mentioned (which I believe needs some TLC too). Will update screenshots to reflect that.

Regarding their usage, I think we should look at consistency first, our opinions about them second. The action menu features them (same popup UI) + events are in a bordered table in the event page. Wether or not it is the right treatment is certainly debatable and I'd like to wait for Design to to chime in before digging deeper. I usually put forward this kind of decision to try to achieve consistency holistically, rather than introducing a new treatment.

mjac0bs commented:
> I added a fix while you were commenting 

Looks like we have a regression in the padding around the notification list now with those latest changes:
![image](https://github.com/linode/manager/assets/114685994/b79f962d-deda-4516-98d8-63f05d50ed26)


> Regarding their usage, I think we should look at consistency first, our opinions about them second. The action menu features them (same popup UI) + events are in a bordered table in the event page. 

Fair point!

> Also, I am so surprised this is what we're debating considering the absolute over-usage (and kinda useless) of the avatar in those 😆

Ha, speaking of, have you seen M3-5818 in the backlog? I have lots of thoughts and opinions about that one which can be summarized as: I'm glad it's stayed in the backlog. 😅

abailly-akamai commented:
@mjac0bs  😮**AGREED**😮

and yes, will fix, ty!

abailly-akamai commented:
@hana-linode & @mjac0bs la nuit porte conseil -  I think that if we're debating this on here, the design change belongs in its own ticket and needs to be handled outside of this refactor if at all. I'll revert the border change, fix the notification issue, update the screenshots and re-request reviews from you two when done. Thanks for the nudges!

mjac0bs reviewed (commented):
Okay, a couple small things. See my comment about the empty state events -- when I check these, I also do check mobile for any regressions since it's a slightly different layout. In this case, it's just the larger breakpoints that have that padding issue.

It also looks like we have some slight misalignment now with the "more" section of the notification list, which was aligned with the "View all events" link before.

This branch:
![Screenshot 2023-05-24 at 8 26 48 AM](https://github.com/linode/manager/assets/114685994/9c1f3a42-fb02-4c65-92d5-33284adb9045)

Prod:
![Screenshot 2023-05-24 at 8 30 38 AM](https://github.com/linode/manager/assets/114685994/463ef031-d452-4524-8340-7473626be92e)

abailly-akamai commented on packages/manager/src/features/NotificationCenter/NotificationSection.tsx:
Thanks for the feedback, so glad you caught these small details - I fixed both so I think we should be ok now

mjac0bs reviewed (approved):
🚢
